### DISCUSSION TITLE
unsticking-the-very-sticky/

### DISCUSSION THREAD

**Discussion by giscus** [zmike/zmike.github.io#55] [Announcements]
# unsticking-the-very-sticky/

Day 4 of Wayland governance hacking
I wake at 5 AM. This is the perfect time to wake up in NYC TZ, as it affords me the ability to eat a whole
apple in the time it takes my little internet-browsing chromebook to load all the IRC and Discord backlogs from
the five hours that I snuck away for a nap when nobody was watching.

https://www.supergoodcode.com/unsticking-the-very-sticky/

**Comment by konradmb** [comment]
🔥🔥🔥**LET HIM COOK**🔥🔥🔥
🗣️📣**THIS IS THE YEAR OF WAYLAND™ DESKTOP**🗣️📣

**Comment by rijnhard** [comment]
Brain dump.

The one thing I think still isn't addressed in the "perfect protocol" problem. 

So wayland has design principles, which is great, but maybe it can stand to use some decision guidelines similar to the Linux kernel.

E.g. For a protocol to be acceptable (including v2+ changes) it doesn't have to address all cases, it should however be valuable in it's current form and have no conceivable way where future use cases could NEVER be addressed. I'd classify that as a hack. So basically if it does the job and doesn't pigeon hole the protocol then ship it.

For integrations on existing protocols, E.g say now existing protocol X has problems A, B & C, and the v2 protocol only solves A but doesn't preclude B & C in the future, then ship it.

I think one of the core concepts of frogs protocols is to be more incremental, which I do t think you have quite hit, experimental protocols are great for discovery and design in early phases but it's not for stable iteration.

**Comment by alphabitserial** [comment]
I think these are all sensible changes.  I was looking through the MR threads for some of the protocols which have been blocked the longest these past days, and noticed a pattern of the same few people — I won't say who — constantly bringing up hypothetical "this protocol's phrasing *technically* allows for someone to implement a compositor in a way nobody is actually going to" and "what if this only solves 99% of cases" type comments.

Even if those hypothetical concerns are true, so what? Ship something! Bug reports **will** come in saying "Hey, you missed my real-world use case," and that's great - it gives version n+1 of the protocol some much-needed direction, motivated by what people actually need!

Thank you for spending the time and effort to improve the status quo, instead of allowing inertia and tradition to hamper forward progress. I sincerely hope these no-nonsense changes to Wayland governance land soon. I'm sure entire ecosystem will benefit tremendously once they do.

**Comment by greydot** [comment]
Congrats! Wayland re-invented both bikeshedding and design by committee. This is not how good designs are done.

**Comment by Channaraj1105** [comment]
We are currently working on a critical training project for our client and have an requirement for a specialized training session on Wayland - WebOS Graphics.
Point of Contact: Channa Raj. S | M: +91 8884040488 | channaraj@vebsters.com

**Comment by youmukonpaku1337** [comment]
what chromebook, and is it running pmos? 🥺
